src/components/app/App.js

```diff
@@ -71,9 +73,32 @@ export default function App({ initialActivities }) {
     </ThemeProvider>
   )
 
+  function handleBookmark(event, id) {
+    event.stopPropagation()
+
+    const activity = activities.find(card => card.id === id)
+
+    const indexActivities = activities.findIndex(card => card.id === id)
+
+    const newActivities = [
+      ...activities.slice(0, indexActivities),
+      {
+        ...activity,
+        isBookmarked: !activity.isBookmarked,
+      },
+      ...activities.slice(indexActivities + 1),
+    ]
+
+    setActivities(newActivities)
+
+    saveToLocal('localActivities', newActivities)
```

Comment:
You could add a `useEffect` which saves your activities whenever your `activities` state is updated.

Reply:
👍